lib/ansible/plugins/become/sudo.py

```diff
@@ -96,6 +96,8 @@ def build_become_command(self, cmd, shell):
             if flags:  # this could be simplified, but kept as is for now for backwards string matching
                 flags = flags.replace('-n', '')
             prompt = '-p "%s"' % (self.prompt)
+        else:
```

Comment:
This seems unrelated to the issue being fixed?

Reply:
we also saw that if password presence changed between loop items the prompt would be incorrectly preserved.